Neos.Flow/Classes/Mvc/Routing/Dto/UriConstraints.php

```diff
@@ -164,6 +166,19 @@ public function withPath(string $path): self
         return new static($newConstraints);
     }
 
+    /**
+     * Create a new instance with the query string constraint added
+     *
+     * @param string $queryString
+     * @return UriConstraints
+     */
+    public function withQueryString(string $queryString): self
+    {
```

Comment:
```suggestion
    {
        $currentQueryString = $this->constraints[self::CONSTRAINT_QUERY_STRING] ?? '';
        if ($queryString === $currentQueryString) {
            return $this;
        }
```
that's roughly what I had in mind btw

Reply:
It's a micro-optimization, but sure, why not?